src/openrct2/ride/Ride.cpp

```diff
@@ -6915,11 +6915,8 @@ void sub_6CB945(Ride* ride)
             if (ride->stations[stationId].Start.xy == RCT_XY8_UNDEFINED)
                 continue;
 
-            LocationXYZ16 location = {
-                (int16_t)(ride->stations[stationId].Start.x * 32),
-                (int16_t)(ride->stations[stationId].Start.y * 32),
-                (ride->stations[stationId].Height),
-            };
+            CoordsXYZ location = { ride->stations[stationId].Start.x * 32, ride->stations[stationId].Start.y * 32,
+                                   ride->stations[stationId].Height * 8 };
             uint8_t direction = 0xFF;
 
             bool specialTrack = false;
```

Comment:
you will need to check the rest of the function for instances of location.z to make sure they take into account the change to it.

Reply:
Solved. I used the ctor of `TileCoordsXYZ(CoordsXYZ)` itself to abstract away the divisions by `8` so that all the logic is on the Location objects themselves.